gerrytools/geometry/optimize.py

```diff
@@ -414,3 +417,29 @@ def calculate_dispersion(units: gpd.GeoDataFrame, enacted_col: str, proposed_col
         raise TypeError("Your enacted and proposed columns must have the same type!")
 
     return units[units[enacted_col] != units[proposed_col]][pop_col].sum()
+
+def calculate_dispersion_per_district(units: gpd.GeoDataFrame, enacted_col: str, proposed_col: str, pop_col: str) -> Dict[int, int]:
+    """ 
+    Calculates dispersion per district in a state given a column with enacted districts
+    and a column with proposed districts. Used in GA.
+    
+    Args:
+        units: The units to optimize on. E.g. census blocks.
+        enacted_col: The column in the GeoDataFrame with the enacted districts.
+        proposed_col: The column in the GeoDataFrame with the proposed districts. 
+        pop_col: The column in the GeoDataFrame with population counts. 
+    
+    Returns:
+        A dictionary with keys as districts, and values as the number of 
+        people displaced from the enacted plan to the proposed plan.
+    """
+
+
+    if units[enacted_col].dtype != units[proposed_col].dtype:
```

Comment:
As discussed, it would probably be better to raise a `TypeError` here rather than silently casting.